src/averageAges.js

```diff
@@ -55,7 +72,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? people.find(mother => mother.name === child.mother) && child.sex === 'm'
+    : people.find(mother => mother.name === child.mother));
+
+  const diffs = children.map(person =>
+    person.born - people.find(mother => mother.name === person.mother).born);
```

Comment:
in general it is quite risky to do such things people.find(mother => mother.name === person.mother).born as find may return undefined. It is better to store in a separate variable and check on validity